frontend/src/routes/FacilityBooking/CreateBooking/index.tsx

```diff
@@ -259,18 +265,19 @@ export default function CreateBooking() {
           />
           <WeeklyRecurrenceRow>
             <StyledTitle>Weekly Recurrence</StyledTitle>
-            <Switch isOn={isWeeklyOn} handleToggle={() => setIsWeeklyOn(!isWeeklyOn)} switchSize={50} />
+            <Switch isOn={isWeeklyOn} handleToggle={() => handleToggleWeekly()} switchSize={50} />
           </WeeklyRecurrenceRow>
           {isWeeklyOn && (
             <SelectableField
               title="End Date"
-              value={bookingEndDate == 0 ? '' : unixToFullDate(bookingEndDate)}
+              value={bookingEndDate === 0 ? '' : unixToFullDate(bookingEndDate)}
               isCompulsory={true}
```

Comment:
```suggestion
              isCompulsory
```
no need explicitely specify true for boolean params